File: fake-store-quarkus/src/main/java/org/arielzarate/domain/model/Product.java
package org.arielzarate.domain.model;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Product {
    private Integer id;
    private String title;
    private Double price;
    private String description;
    private String category;
    private String image;
    private Rating rating;

    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    public static class Rating {
        private Double rate;
        private Integer count;
    }
}

File: api-fake-hexagonal/src/main/kotlin/com/arielZarate/fake_api_hexagonal/domain/model/Product.kt
package com.arielZarate.fake_api_hexagonal.domain.model


data class Product(
    val id: Int?=null,
    val title: String,
    val price: Double,
    val description: String,
    val category: String,
    val image: String,
    val rating: Rating
) {
    data class Rating(val rate: Double, val count: Int)
}